packages/core/src/Synchronizer.ts

```diff
@@ -127,6 +127,11 @@ export class Synchronizer extends EventEmitter {
      * How many blocks the synchronizer will query on each poll. Default: `100000`
      */
     public blockRate: number = 10000
+    /**
+     * The genesis block of the provider environment
+     * @dev Allow passing the genesis block as the starting block for querying events
+     */
+    public genesisBlock: number = 0
```

Comment:
It can be `private _genesisBlock` and we use a getter to access the data
e.g.
https://github.com/Unirep/Unirep/blob/2e9e45969cd276695ea6802876b31706f2d015e6/packages/core/src/Synchronizer.ts#L283

So we won't change this value unexpectedly

Reply:
Thank you for the feedback. This, indeed is a better style. I've fixed it!
Looking forward to any other feedback and suggestions.